src/parser/druid/balance/modules/talents/azeritetraits/DawningSun.js

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import { formatNumber, formatPercentage } from 'common/format';
+import { calculateAzeriteEffects } from 'common/stats';
+import HIT_TYPES from 'game/HIT_TYPES';
+import Analyzer from 'parser/core/Analyzer';
+import AzeritePowerStatistic from 'interface/statistics/AzeritePowerStatistic';
+import SpellLink from 'common/SpellLink';
+import AbilityTracker from 'parser/shared/modules/AbilityTracker';
+import StatTracker from 'parser/shared/modules/StatTracker';
+
+const debug = false;
+
+class DawningSun extends Analyzer {
+  static dependencies = {
+    abilityTracker: AbilityTracker,
+    statTracker: StatTracker,
+  };
+  /**
+   * Gives you a buff that increases your solar wrath damage by x for 8 seconds
+   * lets track how much damage you gained from it
+   */
+
+  bonusDamage = 0;
+
+  getAbility = spellId => this.abilityTracker.getAbility(spellId);
+
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTrait(SPELLS.DAWNING_SUN.id);
+    if (!this.active) {
+      return;
+    }
+    this.bonus = this.selectedCombatant.traitsBySpellId[SPELLS.DAWNING_SUN.id]
+      .reduce((total, rank) => {
+        const [ damage ] = calculateAzeriteEffects(SPELLS.DAWNING_SUN.id, rank);
+        debug && this.log(`Rank ${rank}, damage ${damage}`);
+        return total + damage;
+      }, 0);
+  }
+
+  on_byPlayer_damage(event){
+      const spellId = event.ability.guid;
+      const versPerc = this.statTracker.currentVersatilityPercentage;
+      let critMod = 1;
+
+      if(event.hitType === HIT_TYPES.CRIT){
+        critMod = 2;
+      }
+
+      if(this.selectedCombatant.hasBuff(SPELLS.DAWNING_SUN_BUFF.id) && spellId === SPELLS.SOLAR_WRATH_MOONKIN.id){
+        this.bonusDamage += this.bonus * (1 + versPerc) * critMod;
```

Comment:
i am so jealous that you don't have to deal with boss armor for this calculation